aya/src/programs/links.rs

```diff
@@ -253,12 +252,12 @@ impl Link for ProgAttachLink {
     type Id = ProgAttachLinkId;
 
     fn id(&self) -> Self::Id {
-        ProgAttachLinkId(self.prog_fd, self.target_fd, self.attach_type)
+        ProgAttachLinkId(self.prog_fd, self.target_fd.as_raw_fd(), self.attach_type)
```

Comment:
It could and I considered it but my concern is that changing it into an `i32` or something hides that we are doing something that is not foolproof. By using `RawFd` that immediately jumps out to me as "well that fd could have been closed and we need to address this piece of code in the future". File descriptors are only unique while they are open. The file descriptor could have been closed, a different file opened and the kernel could have handed as a file descriptor with the same integer representation as a previous link. I'd prefer not to change this until we come up with a better way to guarantee correctness here because right now it is "most likely but not necessarily" correct and changing it into an integer wouldn't change that fact.